### PR TITLE
added missing arg and improved example clarity

### PR DESCRIPTION
**Pull request by Erotemic**
The previous example was unclear because the output had 32 classes, but the target had only 4 classes. There was also inconsistencies between what the comments were saying versus what was happening. I added variables to hopefully clear some of these things up, and I also added the arg doc for a missing parameter.

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -172,13 +175,15 @@ class NLLLoss2d(NLLLoss):
 
     Examples::
 
-        >>> m = nn.Conv2d(16, 32, (3, 3)).float()
+        >>> N, C = 5, 4
         >>> loss = nn.NLLLoss2d()
         >>> # input is of size N x C x height x width
-        >>> input = autograd.Variable(torch.randn(3, 16, 10, 10))
+        >>> data = Variable(torch.randn(N, 16, 10, 10))
+        >>> m = nn.Conv2d(16, C, (3, 3)).float()
```
Comment (apaszke) [inline torch/nn/modules/loss.py]: can you please remove the `.float()` call? It's unnecessary here

File: torch/nn/modules/loss.py [hunk 2]
```diff
@@ -172,13 +175,15 @@ class NLLLoss2d(NLLLoss):
...
+        >>> input = m(data)
```
Comment (apaszke) [inline torch/nn/modules/loss.py]: It's not really an input. I think it's better to keep it as `loss(m(input), target)`

Comment (Erotemic) [reply torch/nn/modules/loss.py]: done

File: torch/nn/modules/loss.py (hunk 1 above)
Comment (Erotemic) [reply torch/nn/modules/loss.py]: done

**Review by soumith — APPROVED** [review]

**Comment by apaszke** [comment]
Thank you!
